catalog-connectors/airflow/airflow-package-catalog-connector/airflow_package_catalog_connector/airflow_package_catalog_connector.py

```diff
@@ -0,0 +1,245 @@
+            response = requests.get(airflow_package_download_url,
+                                    timeout=30,
+                                    allow_redirects=True)
+            if response.status_code != 200:
+                # download failed. Log error and abort processing
+                self.log.error(f'Download of archive \'{airflow_package_download_url}\' '
+                               'failed. HTTP response code: {response.status_code}')
+                return operator_key_list
+
+            # save downloaded archive
+            archive = str(self.tmp_archive_dir / airflow_package_name)
+            self.log.debug(f'Saving downloaded archive in \'{archive}\' ...')
+            with open(archive, 'wb') as archive_fh:
+                archive_fh.write(response.content)
+
+            # extract archive
+            self.log.debug(f'Extracting Airflow archive \'{archive}\' ...')
+            with zipfile.ZipFile(archive, 'r') as zip_ref:
+                zip_ref.extractall(self.tmp_archive_dir)
+
+            # delete archive
+            self.log.debug(f'Deleting downloaded Airflow archive \'{archive}\' ...')
+            os.remove(archive)
+
+            # Locate Python scripts that are stored in the 'airflow/operators' directory
+            python_scripts = [str(s) for s in self.tmp_archive_dir.glob('airflow/operators/*.py')]
+
+            #
+            # Identify Python scripts that define classes that extend the
+            # airflow.models.BaseOperator class
+            #
+            scripts_with_operator_class = []  # list of str, containing Python script names with operator definitions
+            extends_baseoperator = []  # list of str, containing classes that extend BaseOperator
+            classes_to_analyze = {}
+            imported_operator_classes = []  # list of str, identifying imported operator classes
+            offset = len(str(self.tmp_archive_dir)) + 1
+            script_count = 0  # used for stats collection
+            class_count = 0   # used for stats collection
+            operator_class_count = 0  # used for stats collection
+            # process each Python script ...
+            for script in python_scripts:
+                script_id = script[offset:]
+                if script_id == 'airflow/operators/__init__.py':
+                    continue
+                script_count += 1
+                self.log.debug(f'Parsing \'{script}\' ...')
+                with open(script, 'r') as source_code:
+                    # parse source code
+                    tree = ast.parse(source_code.read())
+                    # identify imports and class definitions
+                    for node in ast.walk(tree):
+                        if isinstance(node, ast.Import):
+                            pass
+                            # for name in node.names:
+                            #    self.log.debug(f'Detected an IMPORT: {name.name}')
```

Comment:
I'm sure these are already on your radar to remove, but just pointing them out in case (same w/ line 131) :)